### PR TITLE
more tweaks to flutter analysis

### PR DESCRIPTION
**Pull request by devoncarew**
A follow up to my previous PR:
- remove i18n library analysis special-casing - it now generates code using `///` comments
- remove explicit `--flutter-repo` cli flags and instead infer it based on the directory we were started in
- move the `.analysis_options` content from in-lined in the dart code to living in the root of the flutter_tools package. This is to help move us towards making the flutter specific analysis config available to more tools. Currently packages that reference the `flutter` lib will pick up the config options via the `_embedder.yaml` file. `flutter_tools` does not have such a reference, so we instead use an `.analysis_options` file in the root of its project.

### CODE REVIEW

**Comment by Hixie** [comment]
How do you analyze the flutter repo if you're not in the flutter repo? It seems bad to require that you be in a particular directory to get that behaviour.

File: packages/flutter_tools/lib/src/commands/analyze.dart [hunk 1]
```diff
@@ -297,36 +290,11 @@ class AnalyzeCommand extends FlutterCommand {
     /// specify analysis options
     /// note that until there is a default "all-in" lint rule-set we need
     /// to opt-in to all desired lints (https://github.com/dart-lang/sdk/issues/25843)
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: maybe move this comment into the file as well

**Comment by devoncarew** [comment]
> How do you analyze the flutter repo if you're not in the flutter repo? It seems bad to require that you be in a particular directory to get that behaviour.

Ah, I just assumed that's how everybody ran it - that's how I've been doing it. It seemed redundant to ask for a --flutter-repo flag when we could just determine it in the tool. I'll back this back out.

File: packages/flutter_tools/lib/src/commands/analyze.dart (hunk 1 above)
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: kk

**Comment by Hixie** [comment]
FWIW, I often just open a terminal (which could be in any random directory, but often is is just $HOME) and run flutter commands there. :-)

**Comment by Hixie** [comment]
Also sometimes I run "flutter analyze" in my project's directory, but want to include the output for the framework because I've also made changes there.

**Comment by devoncarew** [comment]
PTAL (this is hard to merge w/ ongoing changes)

**Comment by Hixie** [comment]
LGTM

**Comment by devoncarew** [comment]
Hmm, will be abandoning this PR and re-doing it. My rebase has turned it into swiss cheese.

**Pull request closed without merging**